plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/BeanMediatorExtFactory.java

```diff
@@ -67,6 +67,43 @@ public Mediator createSpecificMediator(OMElement elem, Properties properties) {
                 // ignore
             }
         }
+        return mediator;
+    }
+
+        public Mediator createSpecificMediatorForValidation(OMElement elem, Properties properties) {
+
+            BeanMediatorExt mediator = new BeanMediatorExt();
+
+            String attributeValue;
+
+            attributeValue = elem.getAttributeValue(new QName(BeanConstants.VAR));
+            if (attributeValue != null) {
+                mediator.setVarName(attributeValue);
+            }
+
+            attributeValue = elem.getAttributeValue(new QName(BeanConstants.ACTION));
+            if (attributeValue != null) {
+                try {
+                    switch (Action.valueOf(attributeValue.toUpperCase())) {
+                    case CREATE:
+                        populateCreateBeanCase(mediator, elem);
+                        break;
+                    case REMOVE:
+                        mediator.setAction(Action.REMOVE);
+                        break;
+                    case SET_PROPERTY:
+                        populateSetPropertyCaseForValidation(mediator, elem);
+                        break;
+                    case GET_PROPERTY:
+                        populateGetPropertyCaseForValidation(mediator, elem);
+                        break;
+                    default:
+                        assert false;
+                    }
+                } catch (IllegalArgumentException e) {
+                    // ignore
```

Comment:
Why this is ignored? Better add why at least.

Reply:
Fixed ...